File: adapters/visited-places/src/main/java/sk/slsp/workshop/mazesolver/adapters/visitedplaces/adapter/VisitedPlacesInMemoryAdapter.java
package sk.slsp.workshop.mazesolver.adapters.visitedplaces.adapter;

import sk.slsp.workshop.mazesolver.Coordinates;
import sk.slsp.workshop.mazesolver.adapters.visitedplaces.port.ClearVisitedPlacesPort;
import sk.slsp.workshop.mazesolver.adapters.visitedplaces.port.IsAlreadyVisitedPort;
import sk.slsp.workshop.mazesolver.adapters.visitedplaces.port.StoreVisitedPlacePort;

import java.util.HashMap;
import java.util.Map;

public class VisitedPlacesInMemoryAdapter implements ClearVisitedPlacesPort, IsAlreadyVisitedPort, StoreVisitedPlacePort {
	
	private final Map<Coordinates, Boolean> storage = new HashMap<>();
	
	
	@Override
	public void clearVisitedPlaces() {
		storage.clear();
	}
	
	
	@Override
	public Boolean isAlreadyVisited(Integer x, Integer y) {
		return storage.containsKey(new Coordinates(x, y));
	}
	
	
	@Override
	public void storeVisitedPoint(Integer x, Integer y) {
		storage.put(new Coordinates(x, y), true);
	}
}

File: adapters/visited-places/src/main/kotlin/sk/slsp/workshop/mazesolver/adapters/visitedplaces/adapter/VisitedPlacesInMemoryAdapter.kt
package sk.slsp.workshop.mazesolver.adapters.visitedplaces.adapter

import sk.slsp.workshop.mazesolver.Coordinates
import sk.slsp.workshop.mazesolver.adapters.visitedplaces.port.ClearVisitedPlacesPort
import sk.slsp.workshop.mazesolver.adapters.visitedplaces.port.IsAlreadyVisitedPort
import sk.slsp.workshop.mazesolver.adapters.visitedplaces.port.StoreVisitedPlacePort

class VisitedPlacesInMemoryAdapter:
	ClearVisitedPlacesPort,
	StoreVisitedPlacePort,
	IsAlreadyVisitedPort
{

	private val storage: MutableMap<Coordinates, Boolean> = HashMap()

	override fun clearVisitedPlaces() {
		storage.clear()
	}

	override fun storeVisitedPoint(x: Int, y: Int) {
		storage[Coordinates(x, y)] = true
	}

	override fun isAlreadyVisited(x: Int, y: Int): Boolean {
		return storage.contains(Coordinates(x, y))
	}

}
